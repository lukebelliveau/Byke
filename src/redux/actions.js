export const types = {
  LOADING_FINISHED: 'LOADING_FINISHED',
  LOADING_STARTED: 'LOADING_STARTED',
  STATIONS_FETCHED: 'STATIONS_FETCHED',
};

const loadingStarted = () => ({
  type: types.LOADING_STARTED,
});

const loadingFinished = () => ({
  type: types.LOADING_FINISHED,
});

const stationsFetched = stations => ({
  type: types.STATIONS_FETCHED,
  payload: stations,
});

export default {
  loadingStarted,
  loadingFinished,
  stationsFetched,
};
