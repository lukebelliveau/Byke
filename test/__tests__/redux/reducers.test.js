import reducer from '../../../src/redux/reducers';
import actions from '../../../src/redux/actions';
import testLocations from '../../testLocations';

test('LOADING_STARTED sets loading to true', () => {
  const state = reducer();

  const loading = reducer(state, actions.loadingStarted());

  expect(loading).toMatchSnapshot();
});

test('LOADING_FINISHED sets loading to false', () => {
  const state = reducer();

  const loading = reducer(state, actions.loadingFinished());

  expect(loading).toMatchSnapshot();
});

test('STATIONS_FETCHED puts stations in state', () => {
  const stations = ['station1', 'station2'];
  const state = reducer();

  const action = actions.stationsFetched(stations);
  const withStations = reducer(state, actions.stationsFetched(stations));

  expect(withStations).toMatchSnapshot();
});

test('PLACES_FETCHED puts locations in state', () => {
  const locations = testLocations.results;
  const initialState = reducer();

  const withLocations = reducer(
    initialState,
    actions.locationsFetched(locations)
  );

  expect(withLocations).toMatchSnapshot();
});

test('TRIP_SET puts trip data into state', () => {
  const trip = { currentLocation: 'cur', destination: 'des' };
  const state = reducer();

  const withTrip = reducer(state, actions.tripSet(trip));

  expect(withTrip).toMatchSnapshot();
});

test('LOCATION_UPDATED updates location in state', () => {
  const location = { latitude: 10, longitude: 10 };
  const state = reducer();

  const withUpdatedLocation = reducer(state, actions.locationUpdated(location));

  expect(withUpdatedLocation).toMatchSnapshot();
});

test('CHANGED SEARCH_TEXT changes searchText in state', () => {
  const text = 'a query';
  const state = reducer();

  const withSearchText = reducer(state, actions.changeSearchText(text));

  expect(withSearchText).toMatchSnapshot();
});

it('handles improper actions', () => {
  const initialState = reducer();

  const stateAfterAction = reducer(initialState, 'an invalid action');

  expect(initialState).toEqual(stateAfterAction);
});
