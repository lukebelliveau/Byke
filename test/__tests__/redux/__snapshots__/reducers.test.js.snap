// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`puts stations in state 1`] = `
Object {
  "isLoading": false,
  "stations": Array [
    "station1",
    "station2",
  ],
}
`;

exports[`sets loading to false 1`] = `
Object {
  "isLoading": false,
  "stations": Array [],
}
`;

exports[`sets loading to true 1`] = `
Object {
  "isLoading": true,
  "stations": Array [],
}
`;
